src/pubd/repository.rs

```diff
@@ -395,13 +403,39 @@ impl RsyncdStore {
         old_dir.push("old");
 
         if current_dir.exists() {
-            fs::rename(&current_dir, &old_dir)?;
+            fs::rename(&current_dir, &old_dir).map_err(|e| {
+                KrillIoError::new(
+                    format!(
+                        "Could not rename rsync dir from '{}' to '{}' while publishing",
+                        current_dir.to_string_lossy(),
+                        old_dir.to_string_lossy()
+                    ),
+                    e,
+                )
+            })?;
         }
 
-        fs::rename(&new_dir, &current_dir)?;
+        fs::rename(&new_dir, &current_dir).map_err(|e| {
+            KrillIoError::new(
+                format!(
+                    "Could not rename rsync dir from '{}' to '{}' while publishing",
```

Comment:
This error message is the same as the one above, should something be added to make them distinct or will it be clear from the different dir names in the two cases?

Reply:
reworded to make it more clear